aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -274,6 +274,18 @@ Tensor empty_like(
 
   auto memory_format = options.memory_format_opt().value_or(MemoryFormat::Preserve);
 
+  if (options.layout() == kSparseCsr && self.is_sparse_csr()) {
+    auto result = at::native::_sparse_csr_tensor_unsafe(
+        self.crow_indices().clone(),
+        self.col_indices().clone(),
+        at::empty(self.values().sizes(), options.layout(kStrided)),
+        self.sizes(),
+        self.scalar_type(),
```

Comment:
Post offline discussion with @IvanYashchuk - This should be `dtype` if provided, else `self.scalar_type()`.